### PR TITLE
Jams more wireless signals.

### PR DESCRIPTION
**Pull request by PowerfulBacon**
## About The Pull Request

Makes radio/signal jammers jam more wireless signals.

 - Radio devices
 - Suit sensor transmitters
 - Remote door devices
 - AI / Borg remote interactions
 - Camera signals (They work on points not connected to station so must be wireless)
 - Airlocks commanded by wireless NTnet signals
 - Pinpointers (Target must be jammed only)
 - Robot transmissions
 - AI Shells

Price of the jammer upped to 5 TC.

## Why It's Good For The Game

Makes the jammer a more effective tool, making it block more wireless signals which makes more sense. Makes it great for killing stealthily since those darn suit sensors will not transmit signals while in range, but also makes it a lot more obvious since silicon will be able to easily detect the effects in the area.

## Changelog
:cl:
tweak: Radio jammer changed to signal jammer
tweak: Radio jammer price increased from 3 TC to 5 TC
tweak: Signal jammer blocks most outgoing wireless signals.
code: Moves jammer check into a proc on atoms, so it can be overridden and used anywhere.
/:cl:

### CODE REVIEW

**Comment by TerryTehCoder** [comment]
Now this I can get behind. Love the PR, just curious. When Camera Signals are jammed, does that prevent the A.I from seeing in that area?(as if the camera were disabled). Or does it only block the cameras transmission to Camera Monitor consoles?

Also, will Cyborg Binary Channel be affected by this at all? I know a good amount of discussions been brought up in the past on Binary and I wasn't sure. 

I also didn't look too heavily into this, but will the Jammer block the Monitoring program from Nanites from transmitting their sensor data to the crew monitor?

**Comment by PowerfulBacon** [comment]
Should block cameras from transmitting to AIs and camera computers.
Will block silicon binary talk.
Jammer will block nanite suit sensors.

**Comment by bloons3** [comment]
Curious as to why camera signals are blocked when stuff like the AI camera eye detector exists.

**Comment by PowerfulBacon** [comment]
Cause its a wireless signal and all the AI camera eye detector does is let you see where the AI is.

**Comment by eeSPee** [comment]
> Cause its a wireless signal and all the AI camera eye detector does is let you see where the AI is.

Reduce the price of AI eye detector since it is outclassed by this superior item?

**Comment by bloons3** [comment]
> > Cause its a wireless signal and all the AI camera eye detector does is let you see where the AI is.
> 
> Reduce the price of AI eye detector since it is outclassed by this superior item?

Now that I think about it, why would you use an AI detector when the jammer just makes a sphere of no visibility around you.

**Comment by PowerfulBacon** [comment]
> > > Cause its a wireless signal and all the AI camera eye detector does is let you see where the AI is.
> > 
> > 
> > Reduce the price of AI eye detector since it is outclassed by this superior item?
> 
> Now that I think about it, why would you use an AI detector when the jammer just makes a sphere of no visibility around you.

5TC vs 1TC

**Comment by PowerfulBacon** [comment]
> > Cause its a wireless signal and all the AI camera eye detector does is let you see where the AI is.
> 
> Reduce the price of AI eye detector since it is outclassed by this superior item?

Reduce it to 0TC?

**Comment by utilbird** [comment]
Looks pretty cool! Although, this makes a formerly stealthy item pretty obvious when it's enabled. Could this be made into a separate, upgraded version of the original item? i.e "broad spectrum jammer"

File: code/game/objects/items/devices/traitordevices.dm [hunk 1]
```diff
@@ -285,3 +285,11 @@ effective or pretty fucking useless.
 	else
 		GLOB.active_jammers -= src
 	update_icon()
+
+/atom/proc/is_jammed()
+	var/turf/position = get_turf(src)
+	for(var/obj/item/jammer/jammer in GLOB.active_jammers)
```
Comment (ike709) [inline code/game/objects/items/devices/traitordevices.dm:291]: Can this be typeless?

**Review by ike709 — DISMISSED** [review]

File: code/game/objects/items/devices/traitordevices.dm (hunk 1 above)
Comment (PowerfulBacon) [reply code/game/objects/items/devices/traitordevices.dm:291]: probably

File: code/game/objects/items/devices/traitordevices.dm [hunk 2]
```diff
@@ -285,3 +285,11 @@ effective or pretty fucking useless.
...
+		var/turf/jammer_turf = get_turf(jammer)
+		if(position?.z == jammer_turf.z && (get_dist(position, jammer_turf) <= jammer.range))
```
Comment (PowerfulBacon) [inline code/game/objects/items/devices/traitordevices.dm]:
```suggestion
		if(position?.get_virtual_z() == jammer_turf.get_virtual_z() && (get_dist(position, jammer_turf) <= jammer.range))
```

File: code/game/objects/items/devices/traitordevices.dm [hunk 3]
```diff
@@ -285,3 +285,11 @@ effective or pretty fucking useless.
 	else
 		GLOB.active_jammers -= src
 	update_icon()
+
+/atom/proc/is_jammed()
+	var/turf/position = get_turf(src)
+	for(var/obj/item/jammer/jammer in GLOB.active_jammers)
+		var/turf/jammer_turf = get_turf(jammer)
+		if(position?.get_virtual_z() == jammer_turf.get_virtual_z() && (get_dist(position, jammer_turf) <= jammer.range))
```
Comment (PowerfulBacon) [inline code/game/objects/items/devices/traitordevices.dm]:
```suggestion
		if(position?.get_virtual_z_level() == jammer_turf.get_virtual_z_level() && (get_dist(position, jammer_turf) <= jammer.range))
```

**Comment by SergeiKoralev** [comment]
> Looks pretty cool! Although, this makes a formerly stealthy item pretty obvious when it's enabled. Could this be made into a separate, upgraded version of the original item? i.e "broad spectrum jammer"

This is actually a good idea. Also for those of us who don't antag much what is the normal range on a jammer?

**Comment by ike709** [comment]
YOLO

**Pull request merged**